File: BOJ/#1002 터렛/터렛.py
for tc in range(int(input())):
    pos = list(map(int, input().split()))
    dist = ((pos[0]-pos[3])**2 + (pos[1]-pos[4])**2)**0.5

    if dist>abs(pos[2]-pos[5]) and dist<pos[2]+pos[5]: print(2)
    elif dist==abs(pos[2]-pos[5]) or dist==pos[2]+pos[5]:
        if pos[0]==pos[3] and pos[1]==pos[4]: print(-1)
        else: print(1)
    else: print(0)

File: BOJ/#1002 터렛/터렛.cpp
#include <iostream>
#include <string>
#include <cmath>
#include <algorithm>
using namespace std;

int main() {
    int T; cin>>T;
    for(int tc=0; tc<T; tc++) {
        int x1,y1,r1,x2,y2,r2;
        cin>>x1>>y1>>r1>>x2>>y2>>r2;

        // 두 중심의 거리 계산
        double dist = sqrt(pow(x1-x2, 2)+pow(y1-y2, 2));

        // dist가 두 반지름의 차보다 크고 합보다 작으면 두 점에서 만남
        if(dist>abs(r1-r2) && dist<r1+r2) printf("2\n");
        // dist가 두 반지름의 차 or 합과 같으면
        else if(dist==abs(r1-r2) || dist==r1+r2) {
            // 두 원의 중심이 같으면 무한대
            if(x1==x2 && y1==y2) printf("-1\n");
            // 아니면 접함
            else printf("1\n");
        }
        // 둘 다 아니면 만나지 않음
        else printf("0\n");
    }
}
